sunscreen_math/src/metal_impl/ristrettovec.rs

```diff
@@ -0,0 +1,716 @@
+use curve25519_dalek::{edwards::EdwardsPoint, ristretto::RistrettoPoint, CannonicalFieldElement};
+use metal::Buffer;
+
+use core::slice;
+use std::{
+    mem::size_of,
+    ops::{Add, Mul, Sub},
+};
+
+use crate::metal_impl::{ScalarVec, U32Arg};
+
+use super::Runtime;
+
+pub struct RistrettoPointVec {
+    data: Buffer,
+    len: usize,
+}
+
+impl RistrettoPointVec {
+    /// Creates a new [RistrettoVec].
+    ///
+    /// # Remarks
+    /// This code assumes the following layout of curve25519-dalek datastructures:
+    /// ```rust
+    /// struct RistrettoPoint(EdwardsPoint);
+    ///
+    /// struct EdwardsPoint {
+    ///     X: FieldElement2625,
+    ///     Y: FieldElement2625,
+    ///     Z: FieldElement2625,
+    ///     T: FieldElement2625,
+    /// }
+    ///
+    /// struct FieldElement2625([u32; 10]);
+    /// ```
+    /// To achieve this layout, you must use the u32 backend.
+    ///
+    pub fn new(x: &[RistrettoPoint]) -> Self {
+        let runtime = Runtime::get();
+
+        let len = x.len();
+
+        assert_eq!(size_of::<RistrettoPoint>(), size_of::<u32>() * 40);
```

Comment:
Is this not always true? Do we expect the size of a RistrettoPoint to change upstream? If so, it might be better as a debug assert.

Reply:
It is always true, so the compiler can remove this for us. We run our CI and test in release mode, so we want to be alerted if this is ever not true.